pithos/plugins/notification_icon.py

```diff
@@ -91,7 +111,18 @@ def on_has_name_owner(bus, result):
                 self.prepare_complete()
             else:
                 # No fancy tray here
-                self.prepare_complete()
+                logging.info('Checking if org.gnome.Shell is owned and version')
+                Gio.DBusProxy.new(
+                    self.bus,
+                    Gio.DBusProxyFlags.NONE,
+                    None,
+                    'org.gnome.Shell',
```

Comment:
Last time I checked several DE's reported GNOME as the DE in

```python
os.environ['XDG_CURRENT_DESKTOP']
```
As I said sniffing DE's in python isn't reliable.